### PR TITLE
Added compression middleware to enable gzip

### PR DESCRIPTION
**Pull request by bsankara**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -75,7 +76,8 @@ module.exports = {
       app.use(flash());
       app.use(passport.initialize());
       app.use(passport.session());
-
+      app.use(compress());
+
```
Comment (PeteyPii) [inline app.js]: Trailing whitespace here

File: app.js [hunk 2]
```diff
@@ -15,6 +15,7 @@ var localStrat = require('passport-local');
 var Q = require('q');
 var redis = require('redis');
 var favicon = require('serve-favicon');
+var compress = require('compression')
```
Comment (PeteyPii) [inline app.js]: alphabetize by module name

File: package.json [hunk 3]
```diff
@@ -5,6 +5,7 @@
   "dependencies": {
     "bcrypt": "0.8.1",
     "body-parser": "1.12.2",
+    "compression": "^1.6.0",
```
Comment (PeteyPii) [inline package.json]: use an exact version here (remove the caret)

**Comment by PeteyPii** [comment]
Can you also double check that it works, you should be able to see this in the response headers:

https://gyazo.com/0296eec8d5b34387df9001d5964106b6

The reason I say this is because there's some response flush method but I think that might be unrelated to our use case anyways. Would be nice to double-check either way

**Comment by bsankara** [comment]
Don't worry I checked it out, it works, and it doesn't break anything if a user doesn't request compressed pages.

**Comment by PeteyPii** [comment]
Saweeet. Excellent diligence sir. Fix the minor things I noted and I will merge.

**Pull request merged**